src/test/java/com/myFirstProject/myFirstProject/service/AccountServiceImplTest.java

```diff
@@ -0,0 +1,325 @@
+import org.mockito.*;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AccountServiceImplTest {
+
+    @Mock
+    private AccountRepository accountRepository;
+    @Mock
+    private UserRepository userRepository;
+    @Mock
+    private BasketService basketService;
+    @Mock
+    private DeletedBasketService deletedBasketService;
+    @Mock
+    private CurrencyRateService currencyRateService;
+    @Mock
+    private PaymentRepository paymentRepository;
+    //    каптор для того щоб перехопити обєкт який буде виконуватися в методі, в мене використ тому що там
+//    встановл локал дейт нав
+    @Captor
+    private ArgumentCaptor<Payment> paymentArgumentCaptor;
+
+    @Test
+    public void updateAccountIfItNotNull() {
+//        GIVEN
+        MockitoAnnotations.initMocks(this);
+        AccountServiceImpl accountService = new AccountServiceImpl();
+        accountService.setUserRepository(userRepository);
+        PaymentReq paymentReq = buildPaymentReq();
+        User user = buildUser();
+        Mockito.when(userRepository.findById(1l)).thenReturn(Optional.of(user));
+        accountService.setAccountRepository(accountRepository);
+        Account account = buildAccount();
+        Mockito.when(accountRepository.findAccountByUserId(user.getId())).thenReturn(account);
+        accountService.setCurrencyRateService(currencyRateService);
+        Mockito.when(currencyRateService.getRateFromRepository()).thenReturn(buildCurrencyRates());
+        accountService.setPaymentRepository(paymentRepository);
+//        WHEN
+        accountService.updateAccount(paymentReq);
+//        THEN
+        Mockito.verify(paymentRepository).save(paymentArgumentCaptor.capture());
+        Payment payment = paymentArgumentCaptor.getValue();
+        Assert.assertNotNull(payment.getTimeOfPayment());
+        Assert.assertEquals(BigDecimal.TEN, payment.getSum());
+        Assert.assertEquals(account, payment.getAccount());
+        Assert.assertEquals(buildCurrencyEntity(), payment.getCurrencyEntity());
+        Assert.assertEquals(BigDecimal.ONE, payment.getRate());
+        Assert.assertNotNull(account);
+        Assert.assertEquals(BigDecimal.valueOf(11), account.getSum());
+        Assert.assertEquals(2, account.getPayments().size());
+    }
+
+
+    private Account buildAccount() {
+        Account account = new Account();
+        account.setUser(buildUser());
+        account.setSum(BigDecimal.ONE);
+        List<Payment> payments = new ArrayList<>();
+        Payment payment = new Payment();
+        payments.add(payment);
+        account.setPayments(payments);
+        return account;
+    }
+
+    private Account buildNewAccount() {
```

Comment:
please use one method for buildAccount. You should follow DRY principle.

buildAccount(BigDecimal sum)  - use arguments for different fields

Reply:
i created 2 methods because need diferent logic not only sum but and list of payment

Reply:
just add payments as arguments too. 
buildAccount(BigDecimal sum, List<Payments> payments)

Reply:
in 2 case i dont need payments

Reply:
just set null or empty list